### PR TITLE
Feature/prgs feedback april 29

### PR DESCRIPTION
**Pull request by timgaunt**

### CODE REVIEW

File: Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs [hunk 1]
```diff
@@ -40,44 +41,51 @@ public virtual CartRenderingViewModel GetViewModel(string refreshUrl, string rem
...
+            var result = new List<OrderlineViewModel>();
             foreach (var orderLine in basket.OrderLines)
             {
                 var product = CatalogLibrary.GetProduct(orderLine.Sku);
                 var imageService = UCommerce.Infrastructure.ObjectFactory.Instance.Resolve<IImageService>();
                 var orderLineViewModel = new OrderlineViewModel
                 {
                     Quantity = orderLine.Quantity,
-                    ProductName = orderLine.ProductName,
+                    ProductName = product.DisplayName(),
```
Comment (GeorgeAlexandruD) [inline Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs]: I can see it has changed to use the product display name, but after a search of the code, there are two places where orderLine.ProductName is still being used.

ConfirmationEmailModel Line:68
BasketController Line:149

Is this intended? Why is it changed in only some places but not in others?

Also, looked through the code, in AddToBasket Pipeline, the display name is set on the order. So this seems unnecessary.

File: Ucommerce.Sitefinity.UI/Mvc/Model/MiniBasketModel.cs [hunk 2]
```diff
@@ -86,11 +105,11 @@ private int GetNumberOfItemsInBasket()
             }
         }
 
-        private string GetCartPageAbsoluteUrl(Guid cartPageId)
+        private string GettPageAbsoluteUrl(Guid pageId)
```
Comment (GeorgeAlexandruD) [inline Ucommerce.Sitefinity.UI/Mvc/Model/MiniBasketModel.cs]: GetPageAbsoluteURL

File: Ucommerce.Sitefinity.UI/assets/src/js/widgets/review-list.js [hunk 3]
```diff
@@ -60,17 +60,25 @@ function initReviewList(rootElement) {
...
                 this.$http.get(this.purl + '/reviews/data?productId=' + this.pid, {}).then((response) => {
-                    if (response.data &&
-                        response.data.Status &&
+                    if (this.hasValue(response.data) &&
+                        this.hasValue(response.data.Status) &&
                         response.data.Status == 'success' &&
-                        response.data.Data &&
-                        response.data.Data.data &&
-                        response.data.Data.data.Reviews) {
+                        this.hasValue(response.data.Data) &&
```
Comment (GeorgeAlexandruD) [inline Ucommerce.Sitefinity.UI/assets/src/js/widgets/review-list.js]: I can see there is a group of nested data.

Where does it come from and is it possible to flatten it or remodel it for more clarity?

File: Quantum/Mvc/Views/Address/Index.cshtml.orig [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+@model Ucommerce.Sitefinity.UI.Mvc.ViewModels.AddressRenderingViewModel
+
+@Html.StyleSheet(Url.WidgetContent("assets/dist/css/checkout.css"), "head")
+
+@{
+    Layout = null;
+    var showShippingAddress = Model.IsShippingAddressDifferent ? "display:block;" : "display:none;";
+}
+
+<<<<<<< HEAD
```
Comment (GeorgeAlexandruD) [inline Quantum/Mvc/Views/Address/Index.cshtml.orig]: It seems it is not merged or the file is committed by mistake.

File: Ucommerce.Sitefinity.UI/assets/src/js/widgets/products.js [hunk 5]
```diff
@@ -6,9 +6,17 @@ initializeComponent("products", initProducts);
 
 function initProducts(rootElement) {
     const scriptElement = rootElement.querySelector('script[data-variants=true]');
-    const variants = scriptElement === null ? [] : JSON.parse(scriptElement.innerHTML).variants;
+    var variants = [];
+    if (scriptElement) {variants
```
Comment (GeorgeAlexandruD) [inline Ucommerce.Sitefinity.UI/assets/src/js/widgets/products.js]: Dangling "variants" word

File: Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs [hunk 6]
```diff
@@ -159,27 +160,38 @@ public virtual CartUpdateBasketViewModel Update(CartUpdateBasket model)
...
+                        prop.Save();
+                    }
+                }
+            }
+            string voucherPropValue = string.Empty;
+
+            if (basket.Discounts.Count > 0)
+            {
+                var discountCodesList = basket.Discounts.Select(d => d.CampaignItemName).ToList();
+                voucherPropValue = string.Join(",", discountCodesList);
```
Comment (GeorgeAlexandruD) [inline Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs]: Not sure if there is a correlation between joining campaign item names of discounts and setting them as property values.

**Review by GeorgeAlexandruD — CHANGES_REQUESTED** [review]

File: Quantum/Mvc/Views/Address/Index.cshtml.orig (hunk 4 above)
Comment (timgaunt) [reply Quantum/Mvc/Views/Address/Index.cshtml.orig]: I'd imagine so, orig files are usually excluded. I've deleted it and added an ignore

File: Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs (hunk 1 above)
Comment (timgaunt) [reply Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs]: Agreed this isn't needed. It's sometimes done when the cart item has been around for a long time and the admins change names etc but I can't imagine that's the case here. I guess the only other reason was that the product has been retrieved above so it was available. I've removed it anyway and changed to `orderLine.ProductName`

File: Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs (hunk 6 above)
Comment (timgaunt) [reply Ucommerce.Sitefinity.UI/Mvc/Model/CartModel.cs]: Agreed, that should be done in the `RemoveDiscount` logic but I think we'll want to review this block as there's also a block above which I think could be removed:

```
var prop = basket.OrderProperties.FirstOrDefault(v => v.Key == "voucherCodes");
if (prop != null)
{
	prop.Value = prop.Value.Replace(item + ",", string.Empty);
	prop.Save();
}
```

Marking this as closed as I think we'll overhaul aspects of this seperately

File: Ucommerce.Sitefinity.UI/assets/src/js/widgets/review-list.js (hunk 3 above)
Comment (timgaunt) [reply Ucommerce.Sitefinity.UI/assets/src/js/widgets/review-list.js]: `response.data` is the standard JavaScript response.

`response.data.Data` is just a bad name I think, it comes from here: https://github.com/Ucommercenet/Ucommerce.Sitefinity.UI/blob/3aa4b817a5aa8476d59c64c0335feb74f3df59a1/Ucommerce.Sitefinity.UI/Api/Model/OperationStatusDTO.cs#L20 and is used elsewhere in the project so I would suggest leaving that for the time being.

`response.data.Data.data` - this comes from here: https://github.com/Ucommercenet/Ucommerce.Sitefinity.UI/blob/aec6acb797a07fc329c1a35b658b75caba4f9da9/Ucommerce.Sitefinity.UI/Mvc/Controllers/ProductReviewsController.cs#L52 and can be changed without a concern (done)

`response.data.Data.data.Reviews` - comes from the ViewModel as it's just assigning the entire view model. I can't see `Ip` being used so I've just assigned the data to the response. It's still not ideal but at lease it's now `response.data.Data.Reviews`

**Review by GeorgeAlexandruD — APPROVED** [review]

**Pull request merged**